stackstate_checks_base/stackstate_checks/base/utils/http_helper.py

```diff
@@ -0,0 +1,929 @@
+            raise TypeError(message)
+
+    def validate_auth(self):
+        return validate_auth(self._request_model.auth_type,
+                             self._request_model.auth_data)
+
+    def validate_body_type(self):
+        if self._request_model.request_type_validation is None:
+            return True
+
+        # Validate that the `body_type` parameter is a instance of the `HTTPRequestType` enum
+        elif isinstance(self._request_model.request_type_validation, Enum) and \
+                hasattr(self._request_model.request_type_validation, "value") and \
+                self._request_model.request_type_validation.value in self._request_types and \
+                issubclass(self._request_model.request_type_validation.value, type(self._request_model.body)):
+            return True
+
+        else:
+            message = """The body does not conform to the body type provided ({0}).
+                         Either the body type passed to the `validate_body` function needs to be
+                         changed, The validation needs to be removed to allow this body type or
+                         the body needs to be looked at and why it is passing down the incorrect data.
+                         The current body tested content is {1}""".format(
+                str(self._request_model.request_type_validation),
+                str(self._request_model.body))
+            raise ValueError(message)
+
+    def validate_body_schematic(self):
+        if self._request_model.request_schematic_validation is None:
+            return True
+
+        # If the HTTP Response Validation Schematic has been set
+        else:
+            try:
+                # The last part to test is does the Parsed JSON match the schematic validation
+                self._request_model.request_schematic_validation(self._request_model.body) \
+                    .validate()
+                return True
+
+            except DataError as e:
+                message = """The request was unable to conform to the validation schematic.
+                             The error provided by the schematic validation is {0}
+                             To fix this you can either modify the schematic validation or remove it entirely
+                             """.format(e)
+                raise DataError(message)
+
+            except TypeError as e:
+                message = """The request was unable to conform to the validation schematic. The
+                             schematic applied was {0}
+                             The error provided by the schematic validation is {1}
+                             To fix this you can either modify the schematic validation or remove it entirely
+                             """.format(str(self._request_model.request_schematic_validation), e)
+                raise TypeError(message)
+
+    @staticmethod
+    def split_string_into_dict(target, delimiter, sub_delimiter):
```

Comment:
is this still used somewhere?

Reply:
It appears not. I'm removing this forcing the user of the library to send the correct information in the correct fields